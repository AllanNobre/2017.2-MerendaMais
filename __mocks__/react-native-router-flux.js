export const Actions = {
  schoolInfoScreen: () => ({}),
  searchSchool: () => ({}),
  updateInfoScreen: () => ({}),
  profileInfoScreen: () => ({}),
  schedulingVisit: () => ({}),
  manageRegisters: () => ({}),
  lesgislationScreen: () => ({}),
  resolution1Screen: jest.fn(),
  drawerOpen: jest.fn(),
  mainReportsScreen: jest.fn(),
  registerScreen: jest.fn(),
  refresh: jest.fn(),
  StartPendingInspection: jest.fn(),
  scheduleMeeting: jest.fn(),
  scheduleMeetingMap: jest.fn(),

};
