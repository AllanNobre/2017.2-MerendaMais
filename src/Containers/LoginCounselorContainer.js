import { connect } from 'react-redux';
import { LoginConselheiro } from '../screens';
import { modifyCPF, asyncLoginCounselor } from '../actions/counselorActions';

const mapStatetoProps = state => (
  {
    cpf: state.counselor.cpf,
    message_erro: state.counselor.message_erro,
    isLoading: state.counselor.isLoading,
  }
);

export default connect(mapStatetoProps,
  { modifyCPF, asyncLoginCounselor },
)(LoginConselheiro);
