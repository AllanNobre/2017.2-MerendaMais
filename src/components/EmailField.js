import React from 'react';
import PropTypes from 'prop-types';
import { View, TextInput } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import styles from '../Styles';

const EmailField = props => (
  <View style={styles.InputFieldStyle}>
    <MaterialIcons name="email" style={styles.icon} size={26} color="black" />
    <TextInput
      style={styles.InputStyle}
      placeholder="Digite o seu email"
      placeholderTextColor="#95a5a6"
      underlineColorAndroid="transparent"
      returnKeyLabel={'next'}
      maxLength={50}
      keyboardType={'email-address'}
      autoCapitalize={'none'}
      onChangeText={email => props.callback(email)}
      value={props.value}
    />
  </View>
);

EmailField.propTypes = {
  value: PropTypes.string.isRequired,
  callback: PropTypes.func.isRequired,
};

export default EmailField;
