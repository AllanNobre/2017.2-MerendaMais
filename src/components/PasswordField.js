import React from 'react';
import PropTypes from 'prop-types';
import { View, TextInput } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';
import styles from '../Styles';

const changePasswordStyleAccordingToInput = (password) => {
  const passwordRegex = /^(?=.{6,})(?!.*\s).*$/g;

  if (password === '') {
    return styles.InputFieldStyle;
  } else if (passwordRegex.test(password)) {
    return [styles.InputFieldStyle, { borderColor: '#80FF80', borderWidth: 2 }];
  }
  return [styles.InputFieldStyle, { borderColor: '#FF9999', borderWidth: 2 }];
};

const PasswordField = props => (
  <View style={changePasswordStyleAccordingToInput(props.password)}>
    <MaterialIcons name="lock" style={styles.icon} size={26} color="black" />
    <TextInput
      style={styles.InputStyle}
      placeholder="Digite sua senha"
      placeholderTextColor="#95a5a6"
      underlineColorAndroid="transparent"
      returnKeyLabel={'next'}
      maxLength={30}
      keyboardType={'default'}
      onChangeText={password => props.callback(password)}
      secureTextEntry
    />
  </View>
);

PasswordField.propTypes = {
  callback: PropTypes.func.isRequired,
  password: PropTypes.string.isRequired,
};

export default PasswordField;
