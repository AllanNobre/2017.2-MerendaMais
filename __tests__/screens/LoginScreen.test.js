import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import { TouchableOpacity, TextInput, ActivityIndicator } from 'react-native';
// imported as a connected component!
import LoginCounselorContainer from '../../src/Containers/LoginContainer';
import LoginScreen from '../../src/screens/LoginScreen';
import EmailField from '../../src/components/EmailField';
import PasswordField from '../../src/components/PasswordField';
import ButtonWithActivityIndicator from '../../src/components/ButtonWithActivityIndicator';
import { Actions } from '../../__mocks__/react-native-router-flux';


Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();

const initialState = {
  application: {
    isLoading: false,
    message_erro: '',
  },
};

const store = mockStore(initialState);

jest.mock('react-native-router-flux');

describe('Testing LoginCounselor', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <LoginCounselorContainer />,
      { context: { store } },
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Testing Login Screen Input', () => {
  const wrapper = shallow(<LoginScreen />);

  it('should change state when the text of email input component changes', () => {
    const emailInputComponent = wrapper.find(EmailField).dive().find(TextInput);
    expect(wrapper.state('email')).toEqual('');
    emailInputComponent.simulate('ChangeText', 'test5@test.com');
    expect(wrapper.state('email')).toEqual('test5@test.com');
  });

  it('should change state when the text of Password input component changes', () => {
    const passwordInputComponent = wrapper.find(PasswordField).dive().find(TextInput);
    expect(wrapper.state('password')).toEqual('');
    passwordInputComponent.simulate('ChangeText', '12345678');
    expect(wrapper.state('password')).toEqual('12345678');
  });

  it('should update focus on submit', () => {
    const emailInputComponent = wrapper.find(EmailField).dive().find(TextInput);

    // TODO: Refactor to use simulate
    emailInputComponent.props().onSubmitEditing();
    //

    expect(wrapper.state('focus')).toBeTruthy();
  });

  it('Should change screen when Register button is pressed', () => {
    const registerButton = wrapper.find(TouchableOpacity);
    registerButton.simulate('press');
    expect(Actions.registerScreen.mock.calls.length).toBe(1);
  });
});

describe('Testing LoginScreen On pressed buttons', () => {
  it('Test if LoginCounselor button is pressed', () => {
    const asyncLoginCounselor = (state) => {
      expect(state.email).toEqual('Conselheiro@email.com');
      expect(state.password).toEqual('senha');
    };

    const wrapper = shallow(
      <LoginScreen
        asyncLoginCounselor={asyncLoginCounselor}
      />,
    );

    wrapper.setState({
      email: 'Conselheiro@email.com',
      password: 'senha',
    });

    const touch = wrapper.find(ButtonWithActivityIndicator).dive().find(TouchableOpacity);

    expect(touch.length).toEqual(1);
    touch.simulate('press');
  });

  it('Test if loading spinning exists', () => {
    const myProps = {
      ...initialState.application,
      isLoading: true,
    };

    const wrapper = shallow(<LoginScreen {...myProps} />);

    const buttonAndIndicator = wrapper.find(ButtonWithActivityIndicator).dive();

    expect(buttonAndIndicator.find(ActivityIndicator).length).toEqual(1);
  });

  it('Test if loading spinning does not exist', () => {
    const wrapper = shallow(<LoginScreen {...initialState.application} />);
    const buttonAndIndicator = wrapper.find(ButtonWithActivityIndicator).dive();

    expect(buttonAndIndicator.find(ActivityIndicator).length).toEqual(0);
  });
});
