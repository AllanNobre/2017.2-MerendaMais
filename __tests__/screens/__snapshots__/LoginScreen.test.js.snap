// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LoginCounselor renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginScreen
    asyncLoginCounselor={[Function]}
    isLoading={false}
    message_erro=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          backButton={null}
          subTitle=""
          title="Merenda +"
        />,
        <View
          style={
            Object {
              "backgroundColor": "white",
              "flex": 6,
              "justifyContent": "center",
              "paddingHorizontal": 18,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FAFAFA",
                "borderColor": "gray",
                "borderRadius": 7,
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 2,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "margin": 5,
                  "width": 30,
                }
              }
            />
            <TextInput
              keyboardType="email-address"
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Email"
              returnKeyType="next"
              underlineColorAndroid="transparent"
              value={undefined}
              width={280}
            />
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#FAFAFA",
                "borderColor": "gray",
                "borderRadius": 7,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 30,
                "paddingHorizontal": 2,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 30,
                  "margin": 5,
                  "width": 30,
                }
              }
            />
            <TextInput
              onChangeText={[Function]}
              placeholder="Senha"
              returnKeyType="go"
              secureTextEntry={true}
              underlineColorAndroid="transparent"
              value={undefined}
              width={280}
            />
          </View>
          <TouchableOpacity
            activeOpacity={0.7}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FF9500",
                "borderRadius": 8,
                "borderWidth": 1,
                "marginTop": 50,
                "paddingVertical": 18,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              Entrar
            </Text>
          </TouchableOpacity>
        </View>,
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderTopColor": "#a9a9a9",
              "borderTopWidth": 1,
              "flex": 0.5,
              "justifyContent": "center",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.6}
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              Ainda n√£o se cadastrou?
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                 Cadastrar-se
              </Text>
            </Text>
          </TouchableOpacity>
        </View>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "backButton": null,
          "subTitle": "",
          "title": "Merenda +",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FAFAFA",
                  "borderColor": "gray",
                  "borderRadius": 7,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 2,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 30,
                    "margin": 5,
                    "width": 30,
                  }
                }
              />
              <TextInput
                keyboardType="email-address"
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Email"
                returnKeyType="next"
                underlineColorAndroid="transparent"
                value={undefined}
                width={280}
              />
            </View>,
            <View
              style={
                Object {
                  "backgroundColor": "#FAFAFA",
                  "borderColor": "gray",
                  "borderRadius": 7,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 30,
                  "paddingHorizontal": 2,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 30,
                    "margin": 5,
                    "width": 30,
                  }
                }
              />
              <TextInput
                onChangeText={[Function]}
                placeholder="Senha"
                returnKeyType="go"
                secureTextEntry={true}
                underlineColorAndroid="transparent"
                value={undefined}
                width={280}
              />
            </View>,
            <TouchableOpacity
              activeOpacity={0.7}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF9500",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "marginTop": 50,
                  "paddingVertical": 18,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  }
                }
              >
                Entrar
              </Text>
            </TouchableOpacity>,
          ],
          "style": Object {
            "backgroundColor": "white",
            "flex": 6,
            "justifyContent": "center",
            "paddingHorizontal": 18,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 30,
                      "margin": 5,
                      "width": 30,
                    }
                  }
                />,
                <TextInput
                  keyboardType="email-address"
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Email"
                  returnKeyType="next"
                  underlineColorAndroid="transparent"
                  value={undefined}
                  width={280}
                />,
              ],
              "style": Object {
                "alignItems": "center",
                "backgroundColor": "#FAFAFA",
                "borderColor": "gray",
                "borderRadius": 7,
                "borderWidth": 1,
                "flexDirection": "row",
                "paddingHorizontal": 2,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "source": 1,
                  "style": Object {
                    "height": 30,
                    "margin": 5,
                    "width": 30,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "keyboardType": "email-address",
                  "onChangeText": [Function],
                  "onSubmitEditing": [Function],
                  "placeholder": "Email",
                  "returnKeyType": "next",
                  "underlineColorAndroid": "transparent",
                  "value": undefined,
                  "width": 280,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 30,
                      "margin": 5,
                      "width": 30,
                    }
                  }
                />,
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Senha"
                  returnKeyType="go"
                  secureTextEntry={true}
                  underlineColorAndroid="transparent"
                  value={undefined}
                  width={280}
                />,
              ],
              "style": Object {
                "backgroundColor": "#FAFAFA",
                "borderColor": "gray",
                "borderRadius": 7,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginTop": 30,
                "paddingHorizontal": 2,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "source": 1,
                  "style": Object {
                    "height": 30,
                    "margin": 5,
                    "width": 30,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "onChangeText": [Function],
                  "placeholder": "Senha",
                  "returnKeyType": "go",
                  "secureTextEntry": true,
                  "underlineColorAndroid": "transparent",
                  "value": undefined,
                  "width": 280,
                },
                "ref": [Function],
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "LoginCounselor",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.7,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  }
                }
              >
                Entrar
              </Text>,
              "onPress": [Function],
              "style": Object {
                "alignItems": "center",
                "backgroundColor": "#FF9500",
                "borderRadius": 8,
                "borderWidth": 1,
                "marginTop": 50,
                "paddingVertical": 18,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Entrar",
                "disabled": false,
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "white",
                  "fontSize": 20,
                },
              },
              "ref": null,
              "rendered": "Entrar",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <TouchableOpacity
            activeOpacity={0.6}
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              Ainda n√£o se cadastrou?
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                 Cadastrar-se
              </Text>
            </Text>
          </TouchableOpacity>,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderTopColor": "#a9a9a9",
            "borderTopWidth": 1,
            "flex": 0.5,
            "justifyContent": "center",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.6,
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
            >
              Ainda n√£o se cadastrou?
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                 Cadastrar-se
              </Text>
            </Text>,
            "onPress": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": Array [
                "Ainda n√£o se cadastrou?",
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                >
                   Cadastrar-se
                </Text>,
              ],
              "disabled": false,
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": Array [
              "Ainda n√£o se cadastrou?",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": " Cadastrar-se",
                  "disabled": false,
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "blue",
                  },
                },
                "ref": null,
                "rendered": " Cadastrar-se",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            backButton={null}
            subTitle=""
            title="Merenda +"
          />,
          <View
            style={
              Object {
                "backgroundColor": "white",
                "flex": 6,
                "justifyContent": "center",
                "paddingHorizontal": 18,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FAFAFA",
                  "borderColor": "gray",
                  "borderRadius": 7,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 2,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 30,
                    "margin": 5,
                    "width": 30,
                  }
                }
              />
              <TextInput
                keyboardType="email-address"
                onChangeText={[Function]}
                onSubmitEditing={[Function]}
                placeholder="Email"
                returnKeyType="next"
                underlineColorAndroid="transparent"
                value={undefined}
                width={280}
              />
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FAFAFA",
                  "borderColor": "gray",
                  "borderRadius": 7,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 30,
                  "paddingHorizontal": 2,
                }
              }
            >
              <Image
                source={1}
                style={
                  Object {
                    "height": 30,
                    "margin": 5,
                    "width": 30,
                  }
                }
              />
              <TextInput
                onChangeText={[Function]}
                placeholder="Senha"
                returnKeyType="go"
                secureTextEntry={true}
                underlineColorAndroid="transparent"
                value={undefined}
                width={280}
              />
            </View>
            <TouchableOpacity
              activeOpacity={0.7}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF9500",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "marginTop": 50,
                  "paddingVertical": 18,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  }
                }
              >
                Entrar
              </Text>
            </TouchableOpacity>
          </View>,
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderTopColor": "#a9a9a9",
                "borderTopWidth": 1,
                "flex": 0.5,
                "justifyContent": "center",
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.6}
              onPress={[Function]}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                Ainda n√£o se cadastrou?
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                >
                   Cadastrar-se
                </Text>
              </Text>
            </TouchableOpacity>
          </View>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "backButton": null,
            "subTitle": "",
            "title": "Merenda +",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FAFAFA",
                    "borderColor": "gray",
                    "borderRadius": 7,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "paddingHorizontal": 2,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 30,
                      "margin": 5,
                      "width": 30,
                    }
                  }
                />
                <TextInput
                  keyboardType="email-address"
                  onChangeText={[Function]}
                  onSubmitEditing={[Function]}
                  placeholder="Email"
                  returnKeyType="next"
                  underlineColorAndroid="transparent"
                  value={undefined}
                  width={280}
                />
              </View>,
              <View
                style={
                  Object {
                    "backgroundColor": "#FAFAFA",
                    "borderColor": "gray",
                    "borderRadius": 7,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "marginTop": 30,
                    "paddingHorizontal": 2,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "height": 30,
                      "margin": 5,
                      "width": 30,
                    }
                  }
                />
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Senha"
                  returnKeyType="go"
                  secureTextEntry={true}
                  underlineColorAndroid="transparent"
                  value={undefined}
                  width={280}
                />
              </View>,
              <TouchableOpacity
                activeOpacity={0.7}
                onPress={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FF9500",
                    "borderRadius": 8,
                    "borderWidth": 1,
                    "marginTop": 50,
                    "paddingVertical": 18,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Entrar
                </Text>
              </TouchableOpacity>,
            ],
            "style": Object {
              "backgroundColor": "white",
              "flex": 6,
              "justifyContent": "center",
              "paddingHorizontal": 18,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Image
                    source={1}
                    style={
                      Object {
                        "height": 30,
                        "margin": 5,
                        "width": 30,
                      }
                    }
                  />,
                  <TextInput
                    keyboardType="email-address"
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="Email"
                    returnKeyType="next"
                    underlineColorAndroid="transparent"
                    value={undefined}
                    width={280}
                  />,
                ],
                "style": Object {
                  "alignItems": "center",
                  "backgroundColor": "#FAFAFA",
                  "borderColor": "gray",
                  "borderRadius": 7,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "paddingHorizontal": 2,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "source": 1,
                    "style": Object {
                      "height": 30,
                      "margin": 5,
                      "width": 30,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "keyboardType": "email-address",
                    "onChangeText": [Function],
                    "onSubmitEditing": [Function],
                    "placeholder": "Email",
                    "returnKeyType": "next",
                    "underlineColorAndroid": "transparent",
                    "value": undefined,
                    "width": 280,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Image
                    source={1}
                    style={
                      Object {
                        "height": 30,
                        "margin": 5,
                        "width": 30,
                      }
                    }
                  />,
                  <TextInput
                    onChangeText={[Function]}
                    placeholder="Senha"
                    returnKeyType="go"
                    secureTextEntry={true}
                    underlineColorAndroid="transparent"
                    value={undefined}
                    width={280}
                  />,
                ],
                "style": Object {
                  "backgroundColor": "#FAFAFA",
                  "borderColor": "gray",
                  "borderRadius": 7,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginTop": 30,
                  "paddingHorizontal": 2,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "source": 1,
                    "style": Object {
                      "height": 30,
                      "margin": 5,
                      "width": 30,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "onChangeText": [Function],
                    "placeholder": "Senha",
                    "returnKeyType": "go",
                    "secureTextEntry": true,
                    "underlineColorAndroid": "transparent",
                    "value": undefined,
                    "width": 280,
                  },
                  "ref": [Function],
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "LoginCounselor",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.7,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    }
                  }
                >
                  Entrar
                </Text>,
                "onPress": [Function],
                "style": Object {
                  "alignItems": "center",
                  "backgroundColor": "#FF9500",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "marginTop": 50,
                  "paddingVertical": 18,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Entrar",
                  "disabled": false,
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                },
                "ref": null,
                "rendered": "Entrar",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <TouchableOpacity
              activeOpacity={0.6}
              onPress={[Function]}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                Ainda n√£o se cadastrou?
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                >
                   Cadastrar-se
                </Text>
              </Text>
            </TouchableOpacity>,
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderTopColor": "#a9a9a9",
              "borderTopWidth": 1,
              "flex": 0.5,
              "justifyContent": "center",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.6,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
              >
                Ainda n√£o se cadastrou?
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "blue",
                    }
                  }
                >
                   Cadastrar-se
                </Text>
              </Text>,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": Array [
                  "Ainda n√£o se cadastrou?",
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "blue",
                      }
                    }
                  >
                     Cadastrar-se
                  </Text>,
                ],
                "disabled": false,
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": Array [
                "Ainda n√£o se cadastrou?",
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "accessible": true,
                    "allowFontScaling": true,
                    "children": " Cadastrar-se",
                    "disabled": false,
                    "ellipsizeMode": "tail",
                    "style": Object {
                      "color": "blue",
                    },
                  },
                  "ref": null,
                  "rendered": " Cadastrar-se",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
